docs/functions/util.rst

```diff
@@ -26,6 +26,34 @@ These functions are designed to be used by themes.
 
    The returned string is a fully escaped terminal formatting sequence.
 
+.. function:: _lp_grep_fields(filename, delimiter, keys...) -> var:lp_grep_fields
+
+   Parse the given filename for one key/value pairs of the form "<key><delimiter><value>"
+   (e.g. "this=that") on each line.
+   Sets an array containing the parsed values, for each key in the same order the function was called.
+
+   .. code-block:: sh
+      :caption: Example of use
+
+      _lp_grep_fields "CMakeCache.txt" "=" "CMAKE_C_COMPILER:FILEPATH" "CMAKE_CXX_COMPILER:FILEPATH"
+      cmake_c_compiler=${lp_grep_fields[_LP_FIRST_INDEX+0]}
+      cmake_cxx_compiler=${lp_grep_fields[_LP_FIRST_INDEX+1]}
```

Comment:
Good example, but these usages need to be protected for users using their shell in `set -u` mode.
```suggestion
      cmake_c_compiler=${lp_grep_fields[_LP_FIRST_INDEX+0]-}
      cmake_cxx_compiler=${lp_grep_fields[_LP_FIRST_INDEX+1]-}
```